app/services/open_library_service.rb

```diff
@@ -0,0 +1,37 @@
+class OpenLibraryService
+  cattr_accessor :base_url_endpoint
```

Comment:
I would use a constant instead of a class accessor, the base_url will not change in runtime! A very good practice is to extract this sort of things to your secrets/envs, this way you can configure it to use different URLs depending on the environment you are.

For example,
```yaml
# config/secrets.yml
development:
  open_library:
    base_url: <%= ENV.fetch("OPEN_LIBRARY_BASE_URL", "http://dev.openlibrary.org") %>

production:
  open_library:
    base_url: <%= ENV["OPEN_LIBRARY_BASE_URL"] %>
```

```ruby
class OpenLibraryService
  BASE_URL = Rails.application.secrets.open_library[:base_url]
end

Reply:
I didn't know that feature :heart_eyes: , it's more elegant to group a bunch of relationated field and use secrets.group[:field]